File: 2463-minimum-recolors-to-get-k-consecutive-black-blocks/2463-minimum-recolors-to-get-k-consecutive-black-blocks.cpp
class Solution {
public:
    int minimumRecolors(string blocks, int k) {
        int i = 0, j = 0;
        int res = INT_MAX;
        int c = 0;

        while (j < k) {
            if (blocks[j] == 'W')
                c++;
            j++;
        }

        i++;
        res = min(res, c);
        while (i < blocks.length() && j < blocks.length()) {
            if (blocks[i - 1] == 'W')
                c--;
            if (blocks[j] == 'W')
                c++;

            res = min(res, c);

            i++;
            j++;
        }
        return res;
    }
};

File: 2463-minimum-recolors-to-get-k-consecutive-black-blocks/2463-minimum-recolors-to-get-k-consecutive-black-blocks.java
class Solution {
    public int minimumRecolors(String blocks, int k) {
        int i = 0, j = 0;
        int res = Integer.MAX_VALUE;
        int c = 0;

        while (j < k) {
            if (blocks.charAt(j) == 'W')
                c++;
            j++;
        }

        res = Math.min(res, c);

        while (j < blocks.length()) {
            if (blocks.charAt(i) == 'W')
                c--;
            if (blocks.charAt(j) == 'W')
                c++;

            res = Math.min(res, c);

            i++;
            j++;
        }
        return res;
    }
}
